File: src-tauri/src/dto/packs.rs
use std::path::PathBuf;

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PackResponseDto {
    pub name: String,
    pub path: PathBuf,
    pub image: Option<PathBuf>,
    pub last_updated: Option<String>,
    pub from_steam_workshop: bool,
}

File: src/utils/dto/schemas/packs.ts
import { z } from 'zod'

export const PackResponseSchema = z.object({
  name: z.string(),
  path: z.string().optional(),
  image: z.string().nullable().optional(),
  lastUpdated: z.string().nullable().optional(),
  fromSteamWorkshop: z.boolean().optional(),
})
